src/engine/Server.cpp

```diff
@@ -920,6 +924,53 @@ void Server::processUpdateImpl(
   // update anyway (The index of the located triples snapshot is part of the
   // cache key).
   cache_.clearAll();
+
+  json response;
+  response["update"] = plannedQuery.parsedQuery_._originalString;
+  response["status"] = "OK";
+  auto warnings = qet.collectWarnings();
+  warnings.emplace(warnings.begin(),
+                   "SPARQL 1.1 Update for QLever is highly experimental.");
+  response["warnings"] = warnings;
+  RuntimeInformationWholeQuery& runtimeInfoWholeOp =
+      qet.getRootOperation()->getRuntimeInfoWholeQuery();
+  RuntimeInformation& runtimeInfo = qet.getRootOperation()->runtimeInfo();
+  response["runtimeInformation"]["meta"] =
+      nlohmann::ordered_json(runtimeInfoWholeOp);
+  response["runtimeInformation"]["query_execution_tree"] =
+      nlohmann::ordered_json(runtimeInfo);
+  response["delta-triples"]["before"] = nlohmann::json(countBefore);
+  response["delta-triples"]["after"] = nlohmann::json(countAfter);
+  response["delta-triples"]["difference"] =
+      nlohmann::json(countAfter - countBefore);
+  response["time"]["planing"] =
+      absl::StrCat(runtimeInfoWholeOp.timeQueryPlanning.count(), "ms");
+  response["time"]["where"] =
+      absl::StrCat(runtimeInfo.totalTime_.count() / 1000, "ms");
+  json updateTime{
+      {"total", absl::StrCat(updateMetadata.triplePreparationTime_.count() +
+                                 updateMetadata.deletionTime_.count() +
+                                 updateMetadata.insertionTime_.count(),
+                             "ms")},
+      {"triplePreparation",
+       absl::StrCat(updateMetadata.triplePreparationTime_.count(), "ms")},
+      {"deletion", absl::StrCat(updateMetadata.deletionTime_.count(), "ms")},
+      {"insertion", absl::StrCat(updateMetadata.insertionTime_.count(), "ms")}};
+  response["time"]["update"] = updateTime;
+  response["time"]["total"] = absl::StrCat(requestTimer.msecs().count(), "ms");
+  for (auto permutation : Permutation::ALL) {
+    response["located-triples"][Permutation::toString(
+        permutation)]["blocks-affected"] =
+        deltaTriples.getLocatedTriplesForPermutation(permutation).numBlocks();
+    auto numBlocks = index_.getPimpl()
+                         .getPermutation(permutation)
+                         .metaData()
+                         .blockData()
+                         .size();
+    response["located-triples"][Permutation::toString(permutation)]
+            ["blocks-total"] = numBlocks;
+  }
```

Comment:
Can you move this into a separate function which only takes the `DeltaTriples` as `const&`, s.t. we can easily reason about "now we are only extracting metadata"?